### ISSUE TITLE
Base rule extension: block-spacing Extending base rule for TS related blocks

### ISSUE THREAD

**Issue by seahindeniz** [typescript-eslint/typescript-eslint#5990]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+base+rule+extension%22) and found none that match my proposal.
- [X] I have searched the [current extension rule list](https://typescript-eslint.io/rules/#extension-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the base rule

https://eslint.org/docs/latest/rules/block-spacing

### Description

I believe the purpose of the base rule, applies for TS too

### Fail

```typescript
interface Shape {x: number}
interface Shape { x: number}
interface Shape {x: number }
interface Shape {x: number; y: number}
```

### Pass

```typescript
interface Shape { x: number }
interface Shape { x: number; y: number }
```

### Additional Info

Actually, having a rule that rejects single line `interface` declarations could work too

**Comment by JoshuaKGoldberg** [comment]
Accepting PRs for an extension rule that applies the same logic as the base block-spacing rule to TypeScript syntax: interfaces, object literal types, and the like. 👍
